### ISSUE TITLE
Not working across all browsers and OSs

### ISSUE THREAD

**Issue by moorepants** [pydy/pydy-viz#113]
We've found this by trying to run the notebooks in the pydy pycon tutorial.

One issue is that if you have a notebook open in a non-default system web browser then run `scene.display()` it tries to open a new tab in the default browser which may or may not be open.

Here are some notes:
## Windows XP 32bit (in virtual box)
- Firefox 27.0.1 (set as default) the canvas does not load.
- Chrome Version 33.0.1750.117 m everything works.
## Windows 7 64 bit (in virtual box)
- Chrome 35 (canary) works
- Internet Explorer 11 works but all css styling is missing.
- Internet Explorer 10 didn't work (but maybe be because tab tried to open in default browser)
- Firefox 27.0 the canvas is not loading
- Firefox 17.0 the canvas is not loading
## Ubuntu 13.10
- Firefox 27.0 works
- Chromium Version 32.0.1700.102 Ubuntu 13.10 (32.0.1700.102-0ubuntu0.13.10.1~20140128.970.1) works
## Windows 7 Professional 64 bit (real machine)
- Firefox 17.0 ESR works
- Internet Explorer 10 seems to require admin permissions to open the visualization server (if you don't have them).
- Internet Explorer 10 opens an unstyled page with no webgl animation. This is to be expected because IE 10 doesn't support webgl.
## OSX Mavericks
- Safari 7 doesn't load the canvas.
- Works on Firefox 27.0.1
- Get lots of JS errors (like socket errors) on Chrome 33.0.1750.152

**Comment by moorepants** [comment]
@campanawanna Add any notes here with regards to what you are trying.

**Comment by gilbertgede** [comment]
So, I tried downloading and installing the source, but I get the following error:

```
gilbertgede@gilbert-powerbook:~/Software/pydy-viz (master)$ python setup.py instsall
Traceback (most recent call last):
  File "setup.py", line 5, in <module>
    import pydy_viz
  File "/Users/gilbertgede/Software/pydy-viz/pydy_viz/__init__.py", line 16, in <module>
    import visualization_frame
ImportError: No module named 'visualization_frame'
```

I'm running Python 3.3, and am using OS X Mavericks. Haven't gotten to trying Chrome yet...

**Comment by moorepants** [comment]
Does `pip install pydy-viz` work?

**Comment by moorepants** [comment]
Can you import the `visualization_frame` module without installing, just in the top level directory, i.e. `import pydy_viz.visualization_frame`?

**Comment by gilbertgede** [comment]
No to pip install pydy_viz:

```
Downloading/unpacking pydy-viz
  Downloading pydy-viz-0.1.0.tar.gz (345kB): 345kB downloaded
  Running setup.py (path:/private/var/folders/9s/pc60vxkx1cdbwf4mcj_n2rrr0000gn/T/pip_build_gilbertgede/pydy-viz/setup.py) egg_info for package pydy-viz
    Traceback (most recent call last):
      File "<string>", line 17, in <module>
      File "/private/var/folders/9s/pc60vxkx1cdbwf4mcj_n2rrr0000gn/T/pip_build_gilbertgede/pydy-viz/setup.py", line 5, in <module>
        import pydy_viz
      File "./pydy_viz/__init__.py", line 16, in <module>
        import visualization_frame
    ImportError: No module named 'visualization_frame'
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):

  File "<string>", line 17, in <module>

  File "/private/var/folders/9s/pc60vxkx1cdbwf4mcj_n2rrr0000gn/T/pip_build_gilbertgede/pydy-viz/setup.py", line 5, in <module>

    import pydy_viz

  File "./pydy_viz/__init__.py", line 16, in <module>

    import visualization_frame

ImportError: No module named 'visualization_frame'

----------------------------------------
Cleaning up...
Command python setup.py egg_info failed with error code 1 in /private/var/folders/9s/pc60vxkx1cdbwf4mcj_n2rrr0000gn/T/pip_build_gilbertgede/pydy-viz
Storing debug log for failure in /Users/gilbertgede/.pip/pip.log
```

No to `import pydy_viz.visualization_frame` from the pydy-viz/pydy_viz directory:

```
In [1]: import pydy_viz.visualization_frame
---------------------------------------------------------------------------
SystemError                               Traceback (most recent call last)
<ipython-input-1-105ca3670ede> in <module>()
----> 1 import pydy_viz.visualization_frame

/Users/gilbertgede/Software/pydy-viz/pydy_viz/__init__.py in <module>()
     14 __version__ = "0.1.0"
     15 
---> 16 import visualization_frame
     17 from visualization_frame import *
     18 __all__.extend(visualization_frame.__all__)

/Users/gilbertgede/Software/pydy-viz/pydy_viz/visualization_frame.py in <module>()
      6 from sympy.physics.mechanics import Point, ReferenceFrame
      7 
----> 8 from .shapes import Shape
      9 
     10 

SystemError: Parent module '' not loaded, cannot perform relative import
```

**Comment by moorepants** [comment]
Are you on python 2.7? Do you have setuptools installed?

**Comment by gilbertgede** [comment]
No, 3.3, and I do have setuptools installed.

**Comment by moorepants** [comment]
I just ran the visualization notebook in Chrome on a macbook air.

I see the pendulum in the screen, but play doesn't seem to make it move.

The JS console shows:

`Failed to load resource: net::ERR_SOCKET_NOT_CONNECTED`

**Comment by moorepants** [comment]
Refreshing seems to get things moving some. Here is another error I get:

```
GET http://localhost:8000/js/lib/jquery-ui.js net::ERR_CONNECTION_RESET localhost/:10
Resource interpreted as Script but transferred with MIME type text/plain: "http://localhost:8000/js/lib/jquery.min.js". localhost/:9
Resource interpreted as Script but transferred with MIME type text/plain: "http://localhost:8000/js/lib/three.min.js". localhost/:11
Resource interpreted as Script but transferred with MIME type text/plain: "http://localhost:8000/js/lib/TrackballControls.js". localhost/:12
Resource interpreted as Stylesheet but transferred with MIME type text/plain: "http://localhost:8000/css/style.css". localhost/:31
Resource interpreted as Stylesheet but transferred with MIME type text/plain: "http://localhost:8000/css/bootstrap.min.css". localhost/:34
Resource interpreted as Script but transferred with MIME type text/plain: "http://localhost:8000/js/lib/bootstrap.min.js". localhost/:16
Resource interpreted as Script but transferred with MIME type text/plain: "http://localhost:8000/data.json". localhost/:20
Resource interpreted as Script but transferred with MIME type text/plain: "http://localhost:8000/js/canvas/initialize.js". localhost/:22
Resource interpreted as Script but transferred with MIME type text/plain: "http://localhost:8000/js/canvas/addObjects.js". localhost/:23
THREE.WebGLRenderer 58 three.min.js:408
THREE.OrthographicCamera {id: 15, name: "", parent: undefined, children: Array[0], up: THREE.Vector3…}
 addObjects.js:55
Object {shape: Object, name: "unnamed", simulation_matrix: Array[300]}
 addObjects.js:121
Object {color: "black", radius: 0.1, type: "Sphere", name: "unnamed"} addObjects.js:122
Object {shape: Object, name: "unnamed", simulation_matrix: Array[300]}
 addObjects.js:121
Object {color: "black", radius: 0.1, type: "Sphere", name: "unnamed"} addObjects.js:122
Object {shape: Object, name: "unnamed", simulation_matrix: Array[300]}
 addObjects.js:121
Object {color: "black", radius: 0.1, type: "Sphere", name: "unnamed"} addObjects.js:122
Object {shape: Object, name: "unnamed", simulation_matrix: Array[300]}
 addObjects.js:121
Object {color: "black", radius: 0.125, type: "Sphere", name: "unnamed"} addObjects.js:122
Object {shape: Object, name: "Lower Leg", simulation_matrix: Array[300]}
 addObjects.js:121
Object {color: "blue", radius: 0.08, length: 0.611, type: "Cylinder", name: "Lower Leg Cylinder"} addObjects.js:122
Object {shape: Object, name: "Upper Leg", simulation_matrix: Array[300]}
 addObjects.js:121
Object {color: "green", radius: 0.08, length: 0.424, type: "Cylinder", name: "Upper Leg Cylinder"} addObjects.js:122
Object {shape: Object, name: "Torso", simulation_matrix: Array[300]}
 addObjects.js:121
Object {color: "red", radius: 0.08, length: 0.61, type: "Cylinder", name: "Torso Cylinder"} addObjects.js:122
7
Auto Update activated addObjects.js:221
GET http://localhost:8000/js/canvas/animate.js net::ERR_CONNECTION_RESET (index):24
Resource interpreted as Image but transferred with MIME type text/plain: "http://localhost:8000/img/glyphicons-halflings-white.png". (index):123
```

**Comment by moorepants** [comment]
Ok, on Mac in Chrome if I reload multiple times and try clearing the cache and cookies I seem to get more and more of the js loaded. The animation ran when enough of the js was loaded, but there seems to be some kind of socket issue where all the js doesn't load over the socket.

**Comment by moorepants** [comment]
So I used the new static html output in PR #117 and the tutorial example runs just fine in Chrome on Mac OSX. So the issues have something to do with Chrome on Mac and the socket server implementation. I'm not sure why this is different in Mac as opposed to Linux and Windows. I wonder why we implemented a server at all. Exporting the static html/css/js files and opening the index.html file in a web browser works fine. It would probably work in an IPython notebook cell too without much work.
